src/compiler/checker.ts

```diff
@@ -19083,6 +19083,69 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return reportedError;
     }
 
+    /**
+     * Assumes `target` type is assignable to the `Iterable` type, if `Iterable` is defined,
+     * or that it's an array-like type, if `Iterable` is not defined.
+     */
+    function elaborateIterableOrArrayLikeTargetElementwise(

```

Comment:
This function was copied from the existing `elaborateElementwise` function, and then modified to work with a target that is iterable or array-like. Some parts of the elaboration were removed because I figured they didn't apply in this specific case, and some were adapted.